core/src/main/scala/com/banno/cosmos4s/RawCosmosContainer.scala

```diff
@@ -79,30 +74,11 @@ object RawCosmosContainer {
         .eval(createQueryOptionsAlways)
         .map(overrides)
         .flatMap { options =>
-          val sqlParams = parameters
-            .map {
-              case (key, value) =>
-                new SqlParameter(key, value)
-            }
-            .toList
-            .asJava
-          val querySpec = new SqlQuerySpec(query, sqlParams)
-          ReactorCore.fluxToStream(
-            Sync[F].delay(
-              container
-                .queryItems(querySpec, options.build(), classOf[JsonNode])
-                .byPage()
-            )
-          )
+          container.query(query, parameters, options)
         }
-        .flatMap { page =>
-          val elements = page.getElements()
-          if (elements == null) Stream.empty
-          else
-            Chunk
-              .iterable(elements.asScala)
-              .traverse(jacksonToCirce(_).as[A])
-              .fold(Stream.raiseError[F] _, Stream.chunk(_))
+        .evalMapFilter {
```

Comment:
This will break up the `Chunk`s that correspond to the pages. Tthat is why we `traverse`d the `Chunk` directly before.

Similar for `handleResultMeta`.

There is `evalMapChunk` that preservers chunks by traversing the chunk, but there is no equivalent for `evalMapFilter`, maybe we need to define that ourselves here?